packages/kbn-monaco/src/esql/lib/ast/validation/validation.test.ts

```diff
@@ -951,6 +1000,51 @@ describe('validation logic', () => {
       []
     );
 
+    testErrorsAndWarnings('from a | eval a=[1, 2, 3]', []);
+    testErrorsAndWarnings('from a | eval a=[true, false]', []);
+    testErrorsAndWarnings('from a | eval a=["a", "b"]', []);
+    testErrorsAndWarnings('from a | eval a=null', []);
+
+    for (const field of fieldTypes) {
+      testErrorsAndWarnings(`from a | eval ${camelCase(field)}Field IS NULL`, []);
+      testErrorsAndWarnings(`from a | eval ${camelCase(field)}Field IS null`, []);
+      testErrorsAndWarnings(`from a | eval ${camelCase(field)}Field is null`, []);
+      testErrorsAndWarnings(`from a | eval ${camelCase(field)}Field is NULL`, []);
+      testErrorsAndWarnings(`from a | eval ${camelCase(field)}Field IS NOT NULL`, []);
+      testErrorsAndWarnings(`from a | eval ${camelCase(field)}Field IS NOT null`, []);
+      testErrorsAndWarnings(`from a | eval ${camelCase(field)}Field IS not NULL`, []);
+    }
+
+    for (const nesting of [1, 2, 3, 4]) {
+      for (const evenOp of ['-', '+']) {
+        for (const oddOp of ['-', '+']) {
+          // This builds a combination of +/- operators
+          // i.e. ---- something, -+-+ something, +-+- something, etc...
+          const unaryCombination = Array(nesting)
+            .fill('- ')
+            .map((_, i) => (i % 2 ? oddOp : evenOp))
+            .join('');
+          testErrorsAndWarnings(`from a | eval ${unaryCombination} numberField`, []);
+          testErrorsAndWarnings(`from a | eval a=${unaryCombination} numberField`, []);
+          testErrorsAndWarnings(`from a | eval a=${unaryCombination} round(numberField)`, []);
+          testErrorsAndWarnings(`from a | eval 1 + ${unaryCombination} numberField`, []);
+          // still valid
+          testErrorsAndWarnings(`from a | eval 1 ${unaryCombination} numberField`, []);
+        }
+      }
+
+      testErrorsAndWarnings(
+        `from a | eval ${Array(nesting).fill('not ').join('')} booleanField`,
+        []
+      );
+    }
```

Comment:
nit: Does it make sense to dedupe `eval` and `where` tests? Not sure

Reply:
Probably int he long run yes. There's are some difference between the two commands, but there's a big overlap.
Maybe a follow up PR with some tests reorg will be helpful here.